### PR TITLE
WIP Feat 3932

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: apps/backend-functions/src/movie.ts [hunk 1]
```diff
@@ -44,17 +44,57 @@ export async function onMovieDelete(
...
+
+  // Delete events
+  const events = await db.collection('events').where('meta.titleId', '==', movie.id).get();
+  events.docs.forEach(d => batch.delete(d.ref));
+
+  // Delete invitations
+  const eventIds = events.docs.map(e => e.id);
+  if (eventIds.length) {
+    const chunk = 10; // max 10 items in "in" queries
+    let i, j, subEventIds;
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/movie.ts:64]: could you type it ?

File: apps/backend-functions/src/movie.ts [hunk 2]
```diff
@@ -44,17 +44,57 @@ export async function onMovieDelete(
...
+    org.wishlist = org.wishlist.filter(movieId => movieId !== movie.id);
+    batch.update(o.ref, org);
+  });
+
+  // Delete events
+  const events = await db.collection('events').where('meta.titleId', '==', movie.id).get();
+  events.docs.forEach(d => batch.delete(d.ref));
+
+  // Delete invitations
+  const eventIds = events.docs.map(e => e.id);
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/movie.ts:61]: Use a 
```typescript
const promises = events.map(e db.collection('invitations').where('docId', '==', e.id)).then(s => s.docs);
const invitations = awat Promise.all(promises);
invitations.flat().forEach(d => batch.delete(d.ref));
```

File: apps/backend-functions/src/movie.ts [hunk 3]
```diff
@@ -44,17 +44,57 @@ export async function onMovieDelete(
...
+      invitations.docs.forEach(d => batch.delete(d.ref));
+    }
+  }
+
+  // Delete sub-collections (distribution rights)
   await removeAllSubcollections(snap, batch);
 
-  const movieAppAccess = Object.keys(movie.storeConfig.appAccess).filter(access => movie.storeConfig.appAccess[access]);
+  // Delete permissions
+  const orgs = await db.collection('orgs').where('id', 'in', movie.orgIds).get();
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/movie.ts:76]: Same as above

File: libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts [hunk 4]
```diff
@@ -101,5 +117,73 @@ export class MovieComponent implements OnInit {
...
+    if (events.length) {
+      output.push(`${events.length} event(s) will be removed.`);
+    }
+
+    const eventIds = events.map(e => e.id);
+    if (eventIds.length) {
+      let i, j, subEventIds, invitationCount = 0;
+      for (i = 0, j = eventIds.length; i < j; i += chunk) {
+        subEventIds = eventIds.slice(i, i + chunk);
+        const invitations = await this.invitationService.getValue(ref => ref.where('docId', 'in', subEventIds));
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts:161]: Same as above

File: libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts [hunk 5]
```diff
@@ -101,5 +117,73 @@ export class MovieComponent implements OnInit {
...
+      if(invitationCount){
+        output.push(`${invitationCount} invitations to events will be removed.`);
+      }
+    }
+
+    if (this.rights.length) {
+      output.push(`${this.rights.length} distribution rights will be removed.`);
+    }
+
+    const orgs = await this.organizationService.getValue(ref => ref.where('id', 'in', movie.orgIds));
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts:174]: same

File: libs/firebase-utils/src/lib/maintenance.ts [hunk 6]
```diff
@@ -9,8 +8,8 @@ import {
 import { loadAdminServices } from './util';
 
 
-const maintenanceRef = (db?: FirebaseFirestore.Firestore) => {
-  if (!db) db = loadAdminServices().db;
+const maintenanceRef = (db?: FirebaseFirestore.Firestore, checkMissingVars = true) => {
```
Comment (GrandSchtroumpf) [inline libs/firebase-utils/src/lib/maintenance.ts:11]: Avoid boolean parameters in a function, prefer config object, it's easier to read/understand when you invoke it : 
```typescript
const maintenanceRef = (db?: FirebaseFirestore.Firestore, { checkMissingVars = true }: MaintenanceOptions) => {
```

File: libs/firebase-utils/src/lib/util.ts [hunk 7]
```diff
@@ -82,9 +82,9 @@ export interface AdminServices {
 
 let ci: admin.app.App;
 
-export function loadAdminServices(): AdminServices {
+export function loadAdminServices(checkMissingVars = true): AdminServices {
```
Comment (GrandSchtroumpf) [inline libs/firebase-utils/src/lib/util.ts:85]: Same as above

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/firebase-utils/src/lib/util.ts (hunk 7 above)
Comment (phpgeekfr) [reply libs/firebase-utils/src/lib/util.ts:85]: I can do it but it requires to update all call `loadAdminServices()` by `loadAdminServices({})`. Your call ;)

File: libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts [hunk 8]
```diff
@@ -101,5 +117,70 @@ export class MovieComponent implements OnInit {
...
+      }
+    });
+  }
+
+  /**
+   * Used to see what will be deleted before actual removal
+   * @param movie 
+   */
+  private async simulateDeletion(movie: Movie) {
+    const chunk = 10; // max 10 items in "in" queries
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/movie/movie.component.ts:142]: remove this

File: libs/firebase-utils/src/lib/util.ts (hunk 7 above)
Comment (GrandSchtroumpf) [reply libs/firebase-utils/src/lib/util.ts:85]:
```typescript
export function loadAdminServices(options: Options = {}): AdminServices {
```

**Review by GrandSchtroumpf — APPROVED** [review]
Nice

**Pull request merged**